File: src/entities/Customer.ts
export interface Customer {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  address: {
    street: string;
    city: string;
    state: string;
    zipCode: string;
    country: string;
  };
  driverLicense: {
    number: string;
    state: string;
    expiryDate: Date;
  };
  dateOfBirth: Date;
  membershipStatus: 'regular' | 'premium' | 'vip';
  registrationDate: Date;
  paymentMethods: PaymentMethod[];
  rentalHistory: string[]; // Array of rental IDs
}

File: app/schemas/customer.py
from pydantic import BaseModel, EmailStr
from typing import List, Optional
from datetime import date
from enum import Enum

class MembershipStatus(str, Enum):
    regular = "regular"
    premium = "premium"
    vip = "vip"

class CustomerBase(BaseModel):
    first_name: str
    last_name: str
    email: EmailStr
    phone: str
    street: str
    city: str
    state: str
    zip_code: str
    country: str
    driver_license_number: str
    driver_license_state: str
    driver_license_expiry: date
    date_of_birth: date
    membership_status: MembershipStatus
    rental_history: List[str] = []

class CustomerCreate(CustomerBase):
    pass

class Customer(CustomerBase):
    id: str
    registration_date: date

    class Config:
        orm_mode = True